internal/storage/rocksdb/store.go

```diff
@@ -161,6 +238,27 @@ func (rdb *rocksDB) Close() error {
 	return nil
 }
 
+func (rdb *rocksDB) PutTTL(key []byte, value []byte, expireTS int64) error {
+	if expireTS > 0 {
+		defer rdb.opts.statsCli.Timing("rocksdb.putTTL.latency.ms", time.Now())
+		dF := ttlDataFormat{
+			ExpiryTS: expireTS,
+			Data:     value,
+		}
+		msgPack, err := msgpack.Marshal(dF)
+		if err != nil {
+			rdb.opts.statsCli.Incr("rocksdb.putTTL.errors", 1)
+			return err
+		}
+		err = rdb.db.PutCF(rdb.opts.writeOpts, rdb.ttlCF, key, msgPack)
```

Comment:
We may have to think this special case, where the user first invokes this method with a key and TTL. Later, the user invokes this method on the same key but without TTL. So we end up storing this key inside both the CFs. What are the implications of this on `get` and `iterate` ?

Reply:
As discussed fixed by issuing a delete followed by put.